pyrokinetics/pyroscan.py

```diff
@@ -68,47 +73,66 @@ def __init__(
 
         if isinstance(parameter_dict, dict):
             self.parameter_dict = parameter_dict
+        else:
+            raise ValueError("PyroScan takes in a dict object")
 
-            pyro_dict = {}
-
-            # Get len of values for each parameter
-            self.value_size = [len(value) for value in self.parameter_dict.values()]
-
-            # Outer product of input dictionaries - could get very large
-            self.outer_product = list(
-                dict(zip(self.parameter_dict, x))
-                for x in product(*self.parameter_dict.values())
-            )
-
-            # Iterate through all runs and create dictionary
-            for run in self.outer_product:
-
-                single_run_name = ""
-                # Param value for each run written accordingly
-                for param, value in run.items():
-                    single_run_name += (
-                        f"{param}{self.value_separator}{value:{self.value_fmt}}"
-                    )
+        # Load in pyroscan json if there
+        if pyroscan_json is not None:
+            if isinstance(pyroscan_json, str):
+                self.pyroscan_json = json.load(open(pyroscan_json))
+                self.value_fmt = self.pyroscan_json["value_fmt"]
+                self.value_separator = self.pyroscan_json["value_separator"]
+                self.parameter_separator = self.pyroscan_json["parameter_separator"]
+                self.parameter_dict = self.pyroscan_json["parameter_dict"]
+                self.file_name = self.pyroscan_json["file_name"]
+                self.base_directory = self.pyroscan_json["base_directory"]
 
-                    single_run_name += self.parameter_separator
+        else:
+            self.pyroscan_json = {
+                "value_fmt": self.value_fmt,
+                "value_separator": self.value_separator,
+                "parameter_separator": self.parameter_separator,
+                "parameter_dict": self.parameter_dict,
+                "file_name": self.file_name,
+                "base_directory": self.base_directory,
+            }
+
+        pyro_dict = {}
+
+        # Get len of values for each parameter
+        self.value_size = [len(value) for value in self.parameter_dict.values()]
+
+        # Outer product of input dictionaries - could get very large
+        self.outer_product = list(
+            dict(zip(self.parameter_dict, x))
+            for x in product(*self.parameter_dict.values())
+        )
```

Comment:
An alternative is to leave this as a generator:

```suggestion
        # Outer product of input dictionaries - could get very large
        self.outer_product = (
            dict(zip(self.parameter_dict, x))
            for x in product(*self.parameter_dict.values())
        )
```

This will then generate the outer product on the fly, rather than storing the whole thing.